src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -300,7 +300,11 @@ public ProcessModuleCollection Modules
                 if (_modules == null)
                 {
                     EnsureState(State.HaveNonExitedId | State.IsLocal);
-                    _modules = ProcessManager.GetModules(_processId);
+                    var newModules = ProcessManager.GetModules(_processId);
+                    if (Interlocked.CompareExchange(ref _modules, newModules, null) != null)
```

Comment:
Why does this need interlocked? Process instances aren't thread safe.

Reply:
I've seen a few usages of `lock` in the source code and since `Interlocked.CompareExchange` is a single `asm` instruction (at least on x64/x86) I assumed that it would be nice to have it.

Reply:
locks are used for things where the implementation introduces concurrency, e.g. where it fires an event that runs on the thread pool and may execute concurrently with user code accessing the same state.  That's not the case here.